packages/wp-now/src/wp-playground-wordpress/is-wordpress-directory.ts

```diff
@@ -0,0 +1,17 @@
+import fs from 'fs-extra';
+import path from 'path';
+
+/**
+ * Checks if the given path is a WordPress directory.
+ *
+ * @param projectPath The path to the project to check.
+ * @returns Is it a WordPress directory?
+ */
+export function isWordPressDirectory(projectPath: string): Boolean {
+	return (
+		fs.existsSync(path.join(projectPath, 'wp-content')) &&
+		fs.existsSync(path.join(projectPath, 'wp-includes')) &&
+		fs.existsSync(path.join(projectPath, 'wp-load.php')) &&
+		!fs.existsSync(path.join(projectPath, 'wp-config.php'))
+	);
+}
```

Comment:
@danielbachhuber , @wojtekn , @adamziel , Here I'm using `wp-config.php` to decide between `WORDPRESS` and `INDEX` mode, as we discussed.

In the future, we could ask the user if they want to use `wp-config.php` configuration or inject the Sqlite plugin.

Let me know if there is any disagreement.

Reply:
If wp-config.php is missing then it should still do WordPress mode. First it will copy wp-config.sample.php to wp-config.php, then it will setup sqlite. Index mode should kick in if it’s not a WordPress directory at all. Perhaps the presence of wp-load.php would be more useful for infering the mode?

Reply:
Agreed. I've tested it after removing the line that checks for `wp-config.php`, and it works as expected for me: when I run it for the directory that includes by blog's code (from which I removed wp-config.php first), it starts wp-now environment with fresh SQLite database, using all my code, with all plugins and themes in place.

Reply:
Actually, if we remove that line, it recreates `wp-config.php` even if it exists. So it works as expected now, as it will copy wp-config.sample.php to wp-config.php, then it will setup sqlite. However, when I stop the environment and run it again, it crashes:

```Trace: TypeError: Cannot read properties of undefined (reading 'request')
    at file:///Volumes/Sites/wordpress-playground/dist/packages/wp-now/main.js:603:30
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at file:///Volumes/Sites/wordpress-playground/dist/packages/wp-now/main.js:610:15
    at processTicksAndRejections (node:internal/process/task_queues:95:5)```

Reply:
> If wp-config.php is missing then it should still do WordPress mode.
Exactly, currently this is the only case where WORDPRESS mode is inferred.

I agree that if `wp-config.php` exists, the mode should still be WORDPRESS but change the logic. Probably asking the user if load the current config or creating the Sqlite integration.